app/actions/item-actions.js

```diff
@@ -19,18 +22,51 @@ let ItemActions = {
     let saved = item.save();
 
     if (saved) {
-      return saved.then(function() {
+      return saved.then(function(item) {
         AppDispatcher.dispatch({
           actionType: 'ADD_ITEM',
           product,
           item
-        });
+        })
       });
     } else {
       return Promise.reject(item);
     }
   },
 
+  itemForType(type, data, parentId) {
+    if (type === SUBITEMS) {
+      return {
+        title: data.title,
+        type: 'task',
+        parent: parentId
+      }
+    }
+  },
+
+  bulkAdd(type, productId, parent, data) {
+    let product = products.get(productId);
+
+    Promise.map(data, (newItem) => {
+      let newItemData = this.itemForType(type, newItem, parent.number);
+      let item = product.createItem(newItemData, { silent: true });
+
+      item.save();
+
+      return item;
+    }).then(function(items) {
+      _.each(items, (item) => {
+        AppDispatcher.dispatch({
```

Comment:
So I havent seen any errors off the back of this - but I can just dispatch one instead